secret_provider/src/file.rs

```diff
@@ -0,0 +1,26 @@
+use crate::error::Error;
+use crate::SecretProvider;
+
+use async_trait::async_trait;
+use serde::de::DeserializeOwned;
+
+/// A secret provider that reads a file from the provided path.
+pub struct FileSecretProvider;
+
+#[async_trait]
+impl SecretProvider for FileSecretProvider {
+    async fn encryption_parameters<T>(&self, file_path: &str) -> Result<T, Error>
```

Comment:
Since we decoupled the encryption from the secret

```suggestion
    async fn secret_data<T>(&self, file_path: &str) -> Result<T, Error>
```

Reply:
Done.